### PR TITLE
chore: update to the autodetected-controls docs for ember. fixes #13590

### LINKED ISSUE #13590: autogenerate-controls docs outdated for ember
**Describe the bug**
The docs regarding the autogeneration of controls is outdated.

**To Reproduce**
Go to https://storybook.js.org/docs/ember/essentials/controls

**Things to be updated**
Some updates to the docs based on my experience with using Storybook for an Ember Addon:
- @storybook/ember-cli-storybook changed the config key form `ember-cli-storybook` to `@storybook/ember-cli-storybook`
- the docgen-json file is generated in `/dist/storybook-docgen/index.json` not in /storybook-docgen/index.json thus changing code snipped path for registration
- metadata: `import { Button } from './Button';` is not ember like, for ember this import is not required
- metadata: due to a bug in `ember-cli-addon-docs-yuidoc` the name of the component my be surprising. At least for my ember addon, it results in a strange path instead of a nice name.

**Used Dependencies**
- ember addon running on ember 3.20
- "@storybook/addon-essentials": "6.1.11",
- "@storybook/addon-links": "6.1.11",
- "@storybook/addon-storysource": "6.1.11",
- "@storybook/ember": "6.1.11",
- "@storybook/ember-cli-storybook": "0.3.0",
- "ember-cli-addon-docs-yuidoc": "1.0.0",

**Additional note**
Also `node_modules/@storybook/addon-docs/dist/frameworks/ember/jsondoc.js:50` does not fit exactly the structure required by Storybook 6.1.11, but that needs to be addressed in a seperate PR.

### PR DESCRIPTION
**Pull request by crixx**
Issue: #13590

## What I did
- Reworked the documentation and fixed the mentioned details

### CODE REVIEW

**Comment by jonniebigodes** [comment]
@crixx thank you for the time and effort you put into the pull request, could you wait a bit while i take a look at this? 

Stay safe

**Comment by crixx** [comment]
@jonniebigodes any news on this?
Also, I opened a PR to fix the docgen-part I mentioned in the issue: https://github.com/storybookjs/storybook/pull/13595

**Comment by jonniebigodes** [comment]
@crixx sorry this "slipped though the cracks" (pardon the bad pun) as i had to shift my focus to other items. I'll check it tomorrow if you're ok with it.

File: docs/essentials/auto-generated-controls/ember.mdx [hunk 1]
```diff
@@ -1,32 +1,30 @@
...
+  '@storybook/ember-cli-storybook': {
+    enableAddonDocsIntegration: true,
+  },
+});
 ```
 
-Storybook uses this to auto-generate the `ArgTypes` for your component based on docgen information created by [ember-cli-addon-docs-yuidoc](https://github.com/ember-learn/ember-cli-addon-docs-yuidoc#documenting-components).
+Now, running the ember-cli server will generate a JSON documentation file at `/dist/storybook-docgen/index.json`. Since generation of this file is tied into the ember-cli build, it will get regenerated every time component files are saved. For details on documenting your components, check out the examples in the addon that powers the generation [ember-cli-addon-docs-yuidoc](https://github.com/ember-learn/ember-cli-addon-docs-yuidoc#documenting-components).
+
+Storybook uses this file to auto-generate the `ArgTypes` for your component based on docgen information created by [ember-cli-addon-docs-yuidoc](https://github.com/ember-learn/ember-cli-addon-docs-yuidoc#documenting-components).
```
Comment (jonniebigodes) [inline docs/essentials/auto-generated-controls/ember.mdx:13]: @crixx content wise it's almost good on my end. I've checked and one thing that stands. Let me elaborate on this. 
While checking this pull request i used an example that served as the code base for the Intro to Storybook tutorial for Ember, located [here](https://www.learnstorybook.com/intro-to-storybook/ember/en/get-started/), up to the Composite Component section to add some additional content to test this out. The thing here is that when running the script `storybook-dev` (`npm run storybook-dev`) which has the following inside:
```json
{
  "scripts":{
    "storybook-dev": "npm-run-all --continue-on-error --parallel start storybook",
   }
}
```
I get into this:
issue-with-doc-gen-ember

Basically it's constantly in a loop until i kill the terminal window.  One thing that struck me is that this only happens in MacOS, but on Windows it ran ok without any issues. The only way that i could bypass this issue in MacOs was first generate a build by running `npm run build` which is basically:

```json
"build": "ember build --environment=production",
```
And then issuing `npm run storybook` and no more errors in the console. 

I can "chalk this up" (pardon the bad pun) based on [this](https://cli.emberjs.com/release/basic-use/#additionalstepsformacandlinuxusers). But if we could adjust the docs and address this and avoid readers having to search for that i call it a win on our end?

What do you think?

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@crixx just one item and this should be good to go. Let me know what your thoughts are on this and we'll go from there.

Stay safe

File: docs/essentials/auto-generated-controls/ember.mdx (hunk 1 above)
Comment (crixx) [reply docs/essentials/auto-generated-controls/ember.mdx:13]: Jap, that is one of the classic Ember beginner's gotchas on Mac OSX. I completely agree that we should add it.

**Comment by crixx** [comment]
I now added the basic beginner gotchas to the install page, but had to introduce some more snippets to also clarify the exact install steps for ember (these are also outlined in detail on the learnstorybook site). How we keep learnstorybook and this install page in sync. is on another page.

**Comment by jonniebigodes** [comment]
@crixx let me take a closer look at the latest commit as it seems that we'll need to do some work on this.

**Comment by jonniebigodes** [comment]
@crixx i've been checking this and i'm going to detail some things that could be done to improve this and get it merged.

After pulling in the  changes that you added with the latest commit and upon "firing up" the frontpage (Storybook website) with it i was faced with the following:

docs-issue

Based on the screenshot you can see that there's somewhat of an issue with the documentation. 

Upon closer look it seems that there's a little bit for improvement. Factoring in your pull request and how to Storybook needs to be installed and setup for Ember here's what i did:

- Updated the install doc (`docs/get-started/install.md`) to the following:
```md
---
title: 'Install Storybook'
---

Inside your existing project's root directory, issue the following to add Storybook through our CLI:

<!-- prettier-ignore-start -->
<FeatureSnippets
  paths={[
   'get-started/installation-command-section/angular.mdx',
   'get-started/installation-command-section/ember.mdx',
   'get-started/installation-command-section/html.mdx',
   'get-started/installation-command-section/marko.mdx',
   'get-started/installation-command-section/mithril.mdx',
   'get-started/installation-command-section/preact.mdx',
   'get-started/installation-command-section/rax.mdx',
   'get-started/installation-command-section/react.mdx',
   'get-started/installation-command-section/riot.mdx',
   'get-started/installation-command-section/svelte.mdx',
   'get-started/installation-command-section/vue.mdx',
   'get-started/installation-command-section/web-components.mdx',
  ]}
/>

<!-- prettier-ignore-end -->

<details>

<summary><code>sb init</code> is not made for empty projects</summary>

Storybook needs to be installed into a project that is already setup with a framework. It will not work on an empty project. There are many ways to bootstrap an app in given framework including:

- 📦 [Create React App](https://reactjs.org/docs/create-a-new-react-app.html)
- 📦 [Vue CLI](https://cli.vuejs.org/)
- 📦 [Ember CLI](https://guides.emberjs.com/release/getting-started/quick-start/)
- Or any other tooling available.

</details>

During its install process, Storybook will look into your project's dependencies and provide you with the best configuration available.

The command above will make the following changes to your local environment:

- 📦 Install the required dependencies.
- 🛠 Setup the necessary scripts to run and build Storybook.
- 🛠 Add the default Storybook configuration.
- 📝 Add some boilerplate stories to get you started.


<!-- This can be as before as the commands are equal

-->

# shell ( replaced the ` with # to prevent issues with formatting)
npm run storybook


<!-- The are ok to delete as they are not really necessary and avoid cluttering and adding additional entropy -->

<!-- prettier-ignore-start -->
<!-- 
<FeatureSnippets
  paths={[
   'get-started/installation-check-install/angular.mdx',
   'get-started/installation-check-install/ember.mdx',
   'get-started/installation-check-install/html.mdx',
   'get-started/installation-check-install/marko.mdx',
   'get-started/installation-check-install/mithril.mdx',
   'get-started/installation-check-install/preact.mdx',
   'get-started/installation-check-install/rax.mdx',
   'get-started/installation-check-install/react.mdx',
   'get-started/installation-check-install/riot.mdx',
   'get-started/installation-check-install/svelte.mdx',
   'get-started/installation-check-install/vue.mdx',
   'get-started/installation-check-install/web-components.mdx',
  ]}
/> -->

<!-- prettier-ignore-end -->

It will start Storybook locally and output the address. Depending on your system configuration, it will automatically open the address in a new browser tab and you'll be greeted by a welcome screen.


<!-- Storybook welcome screen image below (removed to prevent issues with formatting and loading) -->

There are some noteworthy items here:

- A collection of useful links for more in depth configuration and customization options you have at your disposal.
- A second set of links for you to expand your Storybook knowledge and get involved with the ever growing Storybook community.
- A few example stories to get you started.

Now that you installed Storybook successfully, let’s take a look at a story that was written for us.

<details>
<summary><h4>Troubleshooting</h4></summary>

Below is a curated list to get you unblocked while adding Storybook to your project.

<!--The content below is customized for each individual framework. It now "lives" inside each mdx file. This helps focus the troubleshooting based on a specific framework (i.e Ember with the warning)  -->

<!-- You can use the `--type` flag to tell Storybook to configure itself based on the flag.

For instance you can use:

- `--type react` to setup Storybook with the React configuration options.
- `--type vue` to setup Storybook with the Vue configuration options.
- `--type angular` to setup Storybook with the Angular configuration options. -->


<!-- prettier-ignore-start -->

<FeatureSnippets
  paths={[
   'get-started/installation-problems/angular.mdx',
   'get-started/installation-problems/ember.mdx',
   'get-started/installation-problems/html.mdx',
   'get-started/installation-problems/marko.mdx',
   'get-started/installation-problems/mithril.mdx',
   'get-started/installation-problems/preact.mdx',
   'get-started/installation-problems/rax.mdx',
   'get-started/installation-problems/react.mdx',
   'get-started/installation-problems/riot.mdx',
   'get-started/installation-problems/svelte.mdx',
   'get-started/installation-problems/vue.mdx',
   'get-started/installation-problems/web-components.mdx',
  ]}
/>

<!-- prettier-ignore-end -->

If all else fails, try asking for [help](https://storybook.js.org/support)

</details>

```

- The `get-started/installation-command-section` files hold the following, (leaving in ember.mdx and vue.mdx) to get you situated:
```md
<!-- get-started/installation-command-section/vue.mdx -->

#shell (removed the ` to prevent issues with formatting)
# Add Storybook
npx -p @storybook/cli sb init

If you run into issues with your Storybook, check the troubleshooting section below for guidance on how to solve it.
```

```md
<!-- get-started/installation-command-section/ember.mdx -->

#shell  (removed the triple ` to prevent issues with formatting)
# Add Storybook
npx -p @storybook/cli sb init

# Add Ember storybook adapter
ember install @storybook/ember-cli-storybook

If you run into issues with your Storybook, check the troubleshooting section below for guidance on how to solve it.
```
As you can see the "wildcard" (pardon the bad pun) here is Ember  and how to add it. The other ones follow the same pattern as the vue example.

- Updated the files inside `get-started/installation-problems`. Once again i'm going to leave out a example for Vue and Ember to get you some context:
```md
<!-- get-started/installation-problems/vue.mdx -->

- You can also setup Storybook manually through the Storybook CLI. Add the `--type vue` flag when you initialize Storybook in your project.
- Vue 3 is still not supported in Storybook. The team and the community are actively working on getting it released. Follow up the progression in [here](https://github.com/storybookjs/storybook/issues/10654) and help us get it released.
- If you run into a installation problem, check the [README for the Vue framework](../../app/vue/README.md) for additional instructions.
```

```md
<!-- get-started/installation-problems/ember.mdx -->

- You can also setup Storybook manually through the Storybook CLI. Add the `--type ember`  flag when you initialize Storybook in your project.

- If you see the following message during the `@storybook/ember-cli-storybook` installation process:

# shell (replaced triple `with # to prevent issues with formatting)
The ember generate entity-name command requires an entity name to be specified. 

For more details, use ember help.

This is just a warning! If you followed the instructions provided, everything should he properly configured.


- If you run into a installation problem, check the [README for the Ember framework](../../app/ember/README.md) for additional instructions.
```


Let me know if you're ok with the changes and if you need additional help with it. More than glad to guide you through and get the pull request merged.

**Comment by crixx** [comment]
hi @jonniebigodes thanks a lot for testing it. I like your proposal. I force-pushed the changes to the last commit.

Additionally:
- I left the "get-started/installation-command-section/vue.mdx" as it was before, since currently there's not much in and you're probably on a newer branch -> I tried to merge against `next` but was already up-to-date?
- I fine-tuned the sentence `Check that everything worked by running:` to `Depending on your framework, first build your app and then check that everything worked by running:` such that it's more clear, that the app already needs to be built once before it can work...

Hope everything's fine now.

btw: how do you build and start the docs pages locally? I'm still struggling with `.mdx` and how to get a good preview...

**Comment by crixx** [comment]
hi @jonniebigodes anything I can do?

**Comment by jonniebigodes** [comment]
@crixx i'll take another look at it tomorrow (Sunday), i have some real life stuff to take care of.

**Comment by crixx** [comment]
hi @jonniebigodes anything I can do?

**Comment by jonniebigodes** [comment]
No all good @crixx i'll follow up on this as there's some work that is dependent on this to get through. Once again thank you for the time and effort you put into this pull request.

@shilman self merging and will  follow up.

Stay safe

**Pull request merged**